src/libstd/sys/unix/pipe.rs

```diff
@@ -35,9 +35,18 @@ pub fn anon_pipe() -> io::Result<(AnonPipe, AnonPipe)> {
     {
         weak! { fn pipe2(*mut c_int, c_int) -> c_int }
         if let Some(pipe) = pipe2.get() {
-            cvt(unsafe { pipe(fds.as_mut_ptr(), libc::O_CLOEXEC) })?;
-            return Ok((AnonPipe(FileDesc::new(fds[0])),
-                       AnonPipe(FileDesc::new(fds[1]))));
+            // Note that despite calling a glibc function here we may still get
+            // ENOSYS, see
+            // https://github.com/rust-lang/rust/pull/39386#issuecomment-306970390
+            // for more info.
```

Comment:
Can we not just copy the info here? I.e. "...may still get ENOSYS if the kernel is too old for pipe2 as glibc unconditionally defines the pipe2 symbol as a thin syscall wrapper."